code/modules/spells/spell.dm

```diff
@@ -368,6 +368,12 @@
 		return FALSE
 
 	var/mob/living/living_owner = owner
+	var/sigreturn = SEND_SIGNAL(owner, COMSIG_MOB_TRY_INVOKE_SPELL, src, feedback)
+	if(sigreturn & SPELL_INVOCATION_FAIL)
+		return FALSE
+	if(sigreturn & SPELL_INVOCATION_SUCCESS)
+		return TRUE // Skips following checks.
```

Comment:
we can just do this like this:

```suggestion
	switch(SEND_SIGNAL(owner, COMSIG_MOB_TRY_INVOKE_SPELL, src, feedback))
		if(SPELL_INVOCATION_FAIL)
			return FALSE
		if(SPELL_INVOCATION_SUCCESS)
			return TRUE // skips all of the following checks
```

it has approximately the same line-footprint but I believe that this is a bit more verbose and clean, as well as being 0.000001 cpu cycle faster since we don't have to do two `if()` checks (as well as not holding onto a proc-local arg for the entirity of the rest of the proc), just one switch for both potential cases (and any abnormalities are handled exactly the same way).

Reply:
i just noticed that melbert was the one to feed you this code but i still think that for the purposes and wider-scope of this proc that it's still a relatively good solution. ignore this if you truly want to but i'm fond of this